dydx/client.py

```diff
@@ -72,96 +53,187 @@ def _delete(self, *args, **kwargs):
+        :param startingBefore: optional, defaults to now
+        :type startingBefore: ISO-8601 string
+
         :returns: Array of processed fills
-        :raises: Exception
-        '''
-        return self._get('fills', params=params)
 
-    def create_order(self, **data):
+        :raises: DydxAPIError
+        '''
+        if makerAccountOwner is None:
+            makerAccountOwner = self.public_address
+        if makerAccountNumber is None:
+            makerAccountNumber = self.account_number
+
+        return self._get('dex/fills', params=utils.remove_nones({
+            'makerAccountOwner': makerAccountOwner,
+            'makerAccountNumber': makerAccountNumber,
+            'limit': limit,
+            'startingBefore': startingBefore,
+        }))
+
+    def create_order(
+        self,
+        makerMarket,
+        takerMarket,
+        makerAmount,
+        takerAmount,
+        fillOrKill=False
+    ):
         '''
         Create an order
-        :param fillOrKill: required e.g. false
+
+        :param makerMarket: required
+        :type makerMarket: number
+
+        :param takerMarket: required
+        :type takerMarket: number
+
+        :param makerAmount: required
+        :type makerAmount: number
+
+        :param takerAmount: required
+        :type takerAmount: number
+
+        :param fillOrKill: optional, defaults to False
         :type fillOrKill: bool
-        :param order: required
-        :type order: order
+
         :returns: None
-        :raises: Exception
+
+        :raises: DydxAPIError
         '''
-        order = data['order']
-
-        if 'makerMarket' not in order:
-            raise KeyError('makerMarket not specified in order')
-        if 'takerMarket' not in order:
-            raise KeyError('takerMarket not specified in order')
-        if 'makerAmount' not in order:
-            raise KeyError('makerAmount not specified in order')
-        if 'takerAmount' not in order:
-            raise KeyError('takerAmount not specified in order')
-
-        order.makerAccountOwner = self.public_address
-        order.makerAccountNumber = self.account_number
-        order.takerAccountOwner = self.TAKER_ACCOUNT_OWNER
-        order.takerAccountNumber = self.TAKER_ACCOUNT_NUMBER
-        order.expiration = '0'
-        order.salt = str(random.randInt(0, 2**256))
-        order.typedSignature = self.sign_order(data['order'])
-        return self._post('orders', data=data)
-
-    def delete_order(self, orderHash):
+
+        order = {
+            'makerMarket': makerMarket,
+            'takerMarket': takerMarket,
+            'makerAmount': makerAmount,
+            'takerAmount': takerAmount,
+            'makerAccountOwner': self.public_address,
+            'makerAccountNumber': self.account_number,
+            'takerAccountOwner': self.TAKER_ACCOUNT_OWNER,
+            'takerAccountNumber': self.TAKER_ACCOUNT_NUMBER,
+            'expiration': 0,
+            'salt': random.randint(0, 2**256)
```

Comment:
Does it matter if these salts are psuedo-random instead of true random?

Reply:
not really but I'll make sure the seed is based off the time or something just to prevent collisions